cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -24,67 +24,38 @@ private[lagom] object JoinClusterImpl {
       "lagom.cluster.terminate-system-after-member-removed", TimeUnit.MILLISECONDS
     ).millis
 
-    def exitJvm = config.getBoolean("lagom.cluster.exit-jvm-when-system-terminated")
-
     // join self if seed-nodes are not configured in dev-mode,
     // otherwise it will join the seed-nodes automatically
     val cluster = Cluster(system)
     if (cluster.settings.SeedNodes.isEmpty && joinSelf)
       cluster.join(cluster.selfAddress)
 
-    val SUCCESSFUL_EXIT = 0
-    val exitStatus = new AtomicInteger(SUCCESSFUL_EXIT)
-
-    val isExiting = new AtomicBoolean(false)
-
-    if (exitJvm) {
-      // exit JVM when ActorSystem has been terminated
-      system.registerOnTermination {
-        new Thread("Sys-exiting-from-akka-termination") {
-          override def run(): Unit = {
-            if (isExiting.compareAndSet(false, true)) {
-              println("Proceed to JVM shutdown with exit status: " + exitStatus.get())
-              System.exit(exitStatus.get())
-            } else {
-              println("JVM shutdown already handled. Ignore.")
-            }
-          }
-        }
-      }
-    }
```

Comment:
I think that in a Lagom scope, the actor system should never be made responsible to shutting down the JVM. However trigger the process shutdown should fulfill that responsibility.